src/components/grid-list/grid-list.spec.ts

```diff
@@ -0,0 +1,384 @@
+
+          fixture.componentInstance.colspan = 3;
+          fixture.detectChanges();
+          expect(getProp(tile, 'width')).toBe('299.75px');
+        });
+  });
+
+  it('should allow adjustment of tile rowspan', () => {
+    var template = `
+      <md-grid-list cols="1" rowHeight="100px">
+        <md-grid-tile [rowspan]="rowspan"></md-grid-tile>
+      </md-grid-list>
+    `;
+
+    builder.overrideTemplate(TestGridList, template)
+        .createAsync(TestGridList).then((fixture: ComponentFixture<TestGridList>) => {
+          fixture.componentInstance.rowspan = 2;
+          fixture.detectChanges();
+
+          let tile = fixture.debugElement.query(By.directive(MdGridTile));
+          expect(getProp(tile, 'height')).toBe('201px');
+
+          fixture.componentInstance.rowspan = 3;
+          fixture.detectChanges();
+          expect(getProp(tile, 'height')).toBe('302px');
+        });
+  });
+
+  it('should lay out tiles correctly for a complex layout', () => {
+    var template = `
+      <div style="width:400px">
+        <md-grid-list cols="4" rowHeight="100px">
+          <md-grid-tile *ngFor="let tile of tiles" [colspan]="tile.cols" [rowspan]="tile.rows"
+                        [style.background]="tile.color">
+            {{tile.text}}
+          </md-grid-tile>
+        </md-grid-list>
+      </div>
+    `;
+
+    builder.overrideTemplate(TestGridList, template)
+        .createAsync(TestGridList).then((fixture: ComponentFixture<TestGridList>) => {
+          fixture.componentInstance.tiles = [
+            {text: 'One', cols: 3, rows: 1, color: 'lightblue'},
+            {text: 'Two', cols: 1, rows: 2, color: 'lightgreen'},
+            {text: 'Three', cols: 1, rows: 1, color: 'lightpink'},
+            {text: 'Four', cols: 2, rows: 1, color: '#DDBDF1'},
+          ];
+          fixture.detectChanges();
+          let tiles = getTiles(fixture);
+
+          expect(getProp(tiles[0], 'width')).toBe('299.75px');
+          expect(getProp(tiles[0], 'height')).toBe('100px');
+          expect(getProp(tiles[0], 'left')).toBe('0px');
+          expect(getProp(tiles[0], 'top')).toBe('0px');
+
+          expect(getProp(tiles[1], 'width')).toBe('99.25px');
+          expect(getProp(tiles[1], 'height')).toBe('201px');
+          expect(getProp(tiles[1], 'left')).toBe('300.75px');
+          expect(getProp(tiles[1], 'top')).toBe('0px');
+
+          expect(getProp(tiles[2], 'width')).toBe('99.25px');
+          expect(getProp(tiles[2], 'height')).toBe('100px');
+          expect(getProp(tiles[2], 'left')).toBe('0px');
+          expect(getProp(tiles[2], 'top')).toBe('101px');
+
+          expect(getProp(tiles[3], 'width')).toBe('199.5px');
+          expect(getProp(tiles[3], 'height')).toBe('100px');
+          expect(getProp(tiles[3], 'left')).toBe('100.25px');
+          expect(getProp(tiles[3], 'top')).toBe('101px');
+        });
+  });
+});
+
+@Component({
+  selector: 'test-grid-list',
+  template: ``,
+  styles: [``],
```

Comment:
Can you just omit `template` and `styles`, or do they need to be present for the overrides to work?